zebROS_ws/src/ros_control_boilerplate/src/frc_robot_interface.cpp

```diff
@@ -2570,6 +2585,139 @@ void FRCRobotInterface::write(const ros::Time& time, const ros::Duration& period
 		}
 	}
 
+	this->write_tracer_.start_unique("candle");
+	for (size_t candle_id = 0; candle_id < this->num_candles_; candle_id++) {
+		//ROS_INFO_STREAM("Candle_status_0");
+		if (!(this->candle_local_hardwares_[candle_id]))
+			continue;
+		
+		// Get candle, state, and command interfaces
+		auto &candle = this->candles_[candle_id];
+		auto &candle_state = this->candle_state_[candle_id];
+		auto &candle_command = this->candle_command_[candle_id];
+
+		// For each Command property, check if it changed
+		//  if it did change, update the actual CANdle, and update the state
+
+		// Brightness
+		//ROS_INFO_STREAM("Candle_status_1");
+		double brightness;
+		if (candle_command.brightnessChanged(brightness)) {
+			if (safeTalonCall(
+				candle->ConfigBrightnessScalar(brightness),
+				"candle->ConfigBrightnessScalar",
+				candle_state.getDeviceID()
+			)) {
+				ROS_INFO_STREAM("CANdle " << this->candle_names_[candle_id]
+						<< " : Set brightness to " << brightness);
+				candle_state.setBrightness(brightness);
+			} else {
+				candle_command.resetBrightnessChanged();
+			}
+		}
+
+		// Show status LED when active
+		//ROS_INFO_STREAM("Candle_status_2");
+		bool status_led_when_active;
+		if (candle_command.statusLEDWhenActiveChanged(status_led_when_active)) {
+			if (safeTalonCall(
+				candle->ConfigStatusLedState(!status_led_when_active),
+				"candle->ConfigStatusLedState",
+				candle_state.getDeviceID()
+			)) {
+				ROS_INFO_STREAM("CANdle " << this->candle_names_[candle_id]
+						<< " : Set status led when active to " << status_led_when_active);
+				candle_state.setStatusLEDWhenActive(status_led_when_active);
+			} else {
+				candle_command.resetStatusLEDWhenActiveChanged();
+			}
+		}
+
+		// Enabled
+		//ROS_INFO_STREAM("Candle_status_3");
+		bool enabled;
+		if (candle_command.enabledChanged(enabled)) {
+			if (safeTalonCall(
+				candle->configV5Enabled(enabled),
+				"candle->configV5Enabled",
+				candle_state.getDeviceID()
+			)) {
+				ROS_INFO_STREAM("CANdle " << this->candle_names_[candle_id]
+						<< " : Set enabled to " << enabled);
+				candle_state.setEnabled(enabled);
+			} else {
+				candle_command.resetEnabledChanged();
+			}
+		}
+
+		// Animation
+		//ROS_INFO_STREAM("Candle_status_4");
+		CANdleAnimation candle_animation;
+		Animation* animation;
+		if (candle_command.animationChanged(candle_animation)) {
+			ROS_INFO_STREAM("Candle_status_4.1");
+			// Convert from CANdleAnimation to the appropriate CTRE animation class
+			if (candle_animation.class_type == CANdleAnimationClass::BaseStandard) {
+				ROS_INFO_STREAM("Candle_status_4.1.1");
+				BaseStandardAnimation base_animation = candle_convert::convertBaseStandardAnimation(candle_animation);
+				ROS_INFO_STREAM("Candle_status_4.1.2");
+				animation = &base_animation;
```

Comment:
This isn't going to work, or at least not consistently. animation is going to point to base_animation, and base_animation goes out of scope and becomes invalid once leaving the block it is declared in.  
You might get lucky for a while and that memory not be overwritten by something else, but it will break at some point.

An idea for fixing this. Make a 
`std::shared_ptr<Animation>` that's a member var of frc robot interface.
Make convertAnimation return a `shared_ptr<BaseAnimation>`, constructed using something like `std::make_shared<SpecificAnimationType>(arg, arg, arg);`
Assign the results of convertAnimation to the new class member var.
Same with convertBaseTwoAnimation().

Making the shared_ptr holding the animation a member var will make sure it stays in scope. Although looking at the docs, it just has to stay in scope until Animate() is called, so it could also just be a local declared before the if statement.  That is, just make the current *animation var a shared_ptr and proceed like above.

Reply:
Why this works - make_shared will dynamically allocate memory for the object, and that memory will only be deallocated when the last shared_ptr() pointing to it goes out of scope.

Reply:
Also, you might be able to use std::unique_ptr / std::make_unique instead, but that might get weird with returning it from a function.

Reply:
I applied the shared_ptr fix (and it builds). I might experiment with unique pointers tomorrow, but practice has ended today.